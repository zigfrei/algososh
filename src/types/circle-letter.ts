import { ElementStates } from "./element-states";

export interface ICircleLetter {
  letter: string;
  color: ElementStates;
}