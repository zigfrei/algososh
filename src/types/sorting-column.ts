import { ElementStates } from "./element-states";

export interface ISortingColumn {
  index: number;
  color: ElementStates;
}
