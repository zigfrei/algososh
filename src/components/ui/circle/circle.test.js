import renderer from "react-test-renderer";
import { Circle } from "../circle/circle"
import { ElementStates } from "../../../types/element-states";
import "@testing-library/jest-dom/extend-expect";


describe("testing Circle component for render and Match Snapshot", () => {
  it("circle without text", () => {
    const CircleWithoutText = renderer.create(<Circle />);
    expect(CircleWithoutText).toMatchSnapshot();
  });
  it("circle with text", () => {
    const CircleWithText = renderer.create(<Circle letter={"Snapshot"} />);
    expect(CircleWithText).toMatchSnapshot();
  });
  it("circle with text head", () => {
    const CircleWithTextHead = renderer.create(<Circle head={"Snapshot"} />);
    expect(CircleWithTextHead).toMatchSnapshot();
  });
  it("circle with element head", () => {
    const CircleWithElementHead = renderer.create(
      <Circle head={<Circle isSmall={true} />} />
    );
    expect(CircleWithElementHead).toMatchSnapshot();
  });
  it("circle with text tail", () => {
    const CircleWithTextTail = renderer.create(<Circle tail={"Snapshot"} />);
    expect(CircleWithTextTail).toMatchSnapshot();
  });
  it("circle with element tail", () => {
    const CircleWithElementTail = renderer.create(
      <Circle tail={<Circle isSmall={true} />} />
    );
    expect(CircleWithElementTail).toMatchSnapshot();
  });
  it("small circle", () => {
    const SmallCircle = renderer.create(<Circle isSmall={true} />);
    expect(SmallCircle).toMatchSnapshot();
  });
  it("default circle", () => {
    const DefaultCircle = renderer.create(<Circle />);
    expect(DefaultCircle).toMatchSnapshot();
  });
  it("changing circle", () => {
    const ChangingCircle = renderer.create(
      <Circle state={ElementStates.Changing} />
    );
    expect(ChangingCircle).toMatchSnapshot();
  });
  it("modified circle", () => {
    const ModifiedCircle = renderer.create(
      <Circle state={ElementStates.Modified} />
    );
    expect(ModifiedCircle).toMatchSnapshot();
  });
});
