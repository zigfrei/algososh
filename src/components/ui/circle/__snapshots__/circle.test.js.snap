// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Circle component for render and Match Snapshot changing circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot circle with element head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot circle with element tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  small default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`testing Circle component for render and Match Snapshot circle with text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Snapshot
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot circle with text head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Snapshot
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot circle with text tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    Snapshot
  </div>
</div>
`;

exports[`testing Circle component for render and Match Snapshot circle without text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot default circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot modified circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component for render and Match Snapshot small circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
